pypet2bids/pypet2bids/helper_functions.py

```diff
@@ -611,17 +611,42 @@ def get_recon_method(ReconstructionMethodString: str) -> dict:
             # test_dcm2niix4pet.py may. Will be updated when non-conforming data is obtained
             pass  # do nothing, this case shouldn't fire.....
 
-
-    #if iteration_subset_string:
-    #    ReconMethodName = re.sub(iteration_subset_string, "", contents)
-    #else:
-    #    ReconMethodName = contents
-    ReconMethodName = contents
+    if iteration_subset_string:
+        ReconMethodName = re.sub(iteration_subset_string, "", contents)
+    else:
+        ReconMethodName = contents
 
     # cleaning up weird chars at end or start of name
     ReconMethodName = re.sub(r'[^a-zA-Z0-9]$', "", ReconMethodName)
     ReconMethodName = re.sub(r'^[^a-zA-Z0-9]', "", ReconMethodName)
 
+    expanded_name = ""
+    # get the dimension as it's often somewhere in the name
+    dimension = ""
+
+    search_criteria = r'[1-4][D-d]'
+    if re.search(search_criteria, ReconMethodName):
+        dimension = re.search(search_criteria, ReconMethodName)[0]
+
+    # doing some more manipulation of the recon method name to expand it from not so helpful acronyms
+    possible_names = load_pet_bids_requirements_json(pet_reconstruction_metadata_json)['reconstruction_names']
+
+    # we want to sort the possible names by longest first that we don't break up an acronym prematurely
+    sorted_df = pandas.DataFrame(possible_names).sort_values(by='value', key=lambda x: x.str.len(), ascending=False)
+
+    possible_names = []
+    for row in sorted_df.iterrows():
+        possible_names.append({'value': row[1]['value'], 'name': row[1]['name']})
+
+    for name in possible_names:
+        if name['value'] in ReconMethodName:
+            expanded_name += name['name'] + " "
+            ReconMethodName = re.sub(name['value'], "", ReconMethodName)
+
+    if expanded_name != "":
+        ReconMethodName = dimension + " "*len(dimension) + expanded_name.rstrip()
+        ReconMethodName = " ".join(ReconMethodName.split())
+
```

Comment:
This doesn't keep track of the order of recon method name, but I'm not sure if the order that things are listed in matters or not. @CPernet took the time to keep track of the order in his matlab version of this, but I embraced the mantra that perfect is the enemy of the good and decided that these two things are equivalent:

```3D Ordered Subset Expectation Maximization Point-Spread Function modelling Ordinary Poisson ~= 3D Ordered Subset Expectation Maximization Ordinary Poisson Point-Spread Function modelling```

Reply:
tagging @mnoergaard for comment